### ISSUE TITLE
How to adjust training args to adapt our server without many gpus?

### ISSUE THREAD

**Issue by AQA6666** [NVIDIA/Megatron-LM#203]
We are now trying to train a financial t5-large on our  8*A100(40G) server, and we note that the T5 training parameters in the example are as follows：
T5_ARGS="--num-layers 24 \
         --hidden-size 1024 \
         --num-attention-heads 16 \
         --kv-channels 64 \
         --ffn-hidden-size 3072 \
         --encoder-seq-length 512 \
         --decoder-seq-length 128 \
         --max-position-embeddings 512 \
         --lr 0.0001 \
         --lr-decay-iters 990000 \
         --train-iters 2000000 \
         --min-lr 0.00001 \
         --lr-warmup-fraction 0.01 \
         --micro-batch-size 16 \
         --global-batch-size 2048 \
         --vocab-file $VOCAB_FILE \
         --vocab-extra-ids 100 \
         --split 949,50,1 \
         --fp16"
Note that global-batch-size(2048) / micro-batch-size(16) = 128, I think that these args are fit to gpu nums = 128, but we are now trying to train it on a 8 gpus server, How should we adjust args?

**Comment by AQA6666** [comment]
> We are now trying to train a financial t5-large on our 8*A100(40G) server, and we note that the T5 training parameters in the example are as follows： T5_ARGS="--num-layers 24 --hidden-size 1024 --num-attention-heads 16 --kv-channels 64 --ffn-hidden-size 3072 --encoder-seq-length 512 --decoder-seq-length 128 --max-position-embeddings 512 --lr 0.0001 --lr-decay-iters 990000 --train-iters 2000000 --min-lr 0.00001 --lr-warmup-fraction 0.01 --micro-batch-size 16 --global-batch-size 2048 --vocab-file $VOCAB_FILE --vocab-extra-ids 100 --split 949,50,1 --fp16" Note that global-batch-size(2048) / micro-batch-size(16) = 128, I think that these args are fit to gpu nums = 128, but we are now trying to train it on a 8 gpus server, How should we adjust args?

Another question is that the lr=0.001 with batch size = 2048 , seq length = 512, but in t5's paper , lr is 0.01 with batch size = 1024, seq length = 1024, which args are better choice?

**Comment by AQA6666** [comment]
Third question, we will try to pretrain a t5-XL(3B) model after training t5-large, could this args fit to 3B model, or some of them should be adjusted?

**Comment by AQA6666** [comment]
@jaredcasper
